src/validation.c

```diff
@@ -202,7 +202,7 @@ validate_tunnel_grammar(const NetplanParser* npp, NetplanNetDefinition* nd, yaml
         return yaml_error(npp, node, error, "%s: missing or invalid 'mode' property for tunnel", nd->id);
 
     if (nd->tunnel.mode == NETPLAN_TUNNEL_MODE_WIREGUARD) {
-        if (!nd->tunnel.private_key)
+        if (!nd->tunnel.private_key && (nd->backend_settings.private_key_flags & NETPLAN_PRIVATE_KEY_FLAG_AGENT_OWNED) == 0)
```

Comment:
question: Doesn't that apply to all current values: `agent-owned`, `not-saved` and `not-required`? IIUC, we're just checking for `agent-owned` here...

Reply:
yeah that makes sense, all the private key flags suggest that the private key definition is not required in the file...